### ISSUE TITLE
Re-enable Test_runVerifyGHAArtifactImage after generic container is GA

### ISSUE THREAD

**Issue by laurentsimon** [slsa-framework/slsa-verifier#258]

**Comment by asraa** [comment]
> So we should add tests for "normal" repositories,

Should we specify an env var for allowing this as an exception instead of using a workaround hard-coded in code? Otherwise, I can see why this would be confusing for a user. Users will expect example-package to produce great provenance, but it's actually bad because of its references to `main`

**Comment by laurentsimon** [comment]
Why would users expect example-package to generate "great" provenance? Is it because it serves as example repo?
I'm worried with env variables, because they let anyone - not just us - to change the behavior of the verifier in a less secure way.

**Comment by asraa** [comment]
Yeah, I assume because it serves as an example provenance repository.

> they let anyone - not just us - change the behavior of the verifier in a less secure way.

I would not allow the env var to apply to all repos, just the `if exampleRepo && bypassVarEnabled; then skip tag check`

so that even if users applied the env var, they wouldn't be able to reduce security. It's just for example package.

But I think it does complicate things. If it doesn't simplify the test burden here much, then not worries.

**Comment by laurentsimon** [comment]
It would avoid re-generating the examples for the generic container. But there are only a few tests and they use the same image. I'll keep the idea in mind.

**Comment by laurentsimon** [comment]
btw, using an env variable works, but we jut need to make use of `GITHUB_REPOSITORY` instead of creating our own:
```
// The trusted builders needs to build themselves at `@main`.
isBuilderRepoBinary := id.CallerRepository == trustedBuilderRepository
// End-to-end tests always build at `@main`. We only allow `@main` if the
// verification occurs in the end-to-end repository.
isE2eTest := (os.Getenv("GITHUB_REPOSITORY") == e2eTestRepository &&
	id.CallerRepository == e2eTestRepository)
if (isBuilderRepoBinary || isE2eTest) &&
	strings.EqualFold("refs/heads/main", ref) {
	return nil
}
```

**Comment by ianlewis** [comment]
@laurentsimon You mentioned this was completed? can we close this?
